### PR TITLE
Bump viser

### PR DESCRIPTION
**Pull request by ginazhouhuiwu**
Viser api changes and bump version to 0.2.1

### CODE REVIEW

File: nerfstudio/viewer_legacy/viser/server.py
Comment (brentyi) [inline nerfstudio/viewer_legacy/viser/server.py:47]: sorry, I think this is now renamed to: https://viser.studio/latest/infrastructure/#viser.infra.WebsockServer

File: pyproject.toml
Comment (brentyi) [inline pyproject.toml:60]: maybe we can actually bump to `0.2.2` once https://github.com/nerfstudio-project/viser/pull/223 is merged, since that resolves some unlikely but plausible race conditions in the messaging infra!

**Review by brentyi — COMMENTED** [review]
🎊 🎊 🎊

File: pyproject.toml
Comment (ginazhouhuiwu) [reply pyproject.toml]: bumped!

File: nerfstudio/viewer_legacy/viser/server.py
Comment (ginazhouhuiwu) [reply nerfstudio/viewer_legacy/viser/server.py]: change is made

**Review by kerrj — APPROVED** [review]
lgtm, as long as @brentyi doesn't have any other changes

**Comment by brentyi** [comment]
Yeah I think we can go ahead with the bump as soon as these fixes are merged + we do a viser release, otherwise we have some regressions:
https://github.com/nerfstudio-project/viser/pull/223
https://github.com/nerfstudio-project/viser/pull/228
https://github.com/nerfstudio-project/viser/pull/229

**Pull request merged**
